app/services/shorten_url.rb

```diff
@@ -0,0 +1,16 @@
+class ShortenUrl
+
+  SHORTEN_URL_LOG = 'log/tinyurl.log'
+
+  def self.short(url)
+    response = HTTParty.get("http://tinyurl.com/api-create.php?url=#{url}")
+    raise HTTParty::Error.new response if response.match? 'ERROR'
```

Comment:
I'm note sure we'll ever see a response that matches that string.  You should check the response.code against "success" codes, and raise the exception if it is not.

For example, I forced a failure by sending a bad URI:
```
response = HTTParty.get("http://tinyurl.com/api-create.ph?url=''")
=> "<?xml version=\"1.0\" encoding=\"iso-8859-1\"?>\n" +
"<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"\n" +
"         \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">\n" +
"<html xmlns=\"http://www.w3.org/1999/xhtml\" xml:lang=\"en\" lang=\"en\">\n" +
" <head>\n" +
"  <title>404 - Not Found</title>\n" +
" </head>\n" +
" <body>\n" +
"  <h1>404 - Not Found</h1>\n" +
" </body>\n" +
"</html>\n"
[4] pry(main)> response.inspect
=> "#<HTTParty::Response:0x7fd4ae91e850 parsed_response=\"<?xml version=\\\"1.0\\\" encoding=\\\"iso-8859-1\\\"?>\\n<!DOCTYPE html PUBLIC \\\"-//W3C//DTD XHTML 1.0 Transitional//EN\\\"\\n         \\\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\\">\\n<html xmlns=\\\"http://www.w3.org/1999/xhtml\\\" xml:lang=\\\"en\\\" lang=\\\"en\\\">\\n <head>\\n  <title>404 - Not Found</title>\\n </head>\\n <body>\\n  <h1>404 - Not Found</h1>\\n </body>\\n</html>\\n\", @response=#<Net::HTTPNotFound 404 Not Found readbody=true>, @headers={\"date\"=>[\"Sat, 14 Jul 2018 18:03:21 GMT\"], \"content-type\"=>[\"text/html\"], \"transfer-encoding\"=>[\"chunked\"], \"connection\"=>[\"close\"], \"set-cookie\"=>[\"__cfduid=d071b4ebad4a29a20fd2ba0b0f2cd456a1531591401; expires=Sun, 14-Jul-19 18:03:21 GMT; path=/; domain=.tinyurl.com; HttpOnly\"], \"server\"=>[\"cloudflare\"], \"cf-ray\"=>[\"43a5e7d1967f2180-EWR\"]}>"
[5] pry(main)> 
[6] pry(main)> response.code
=> 404
[7] pry(main)> response.message
=> "Not Found"
```
In `hips_service.rb`, for example, we do this:
```
SUCCESS_CODES = [200, 201, 202].freeze
.
.
.
parsed_response = response.parsed_response

return parsed_response if SUCCESS_CODES.include?(response.code)
.
<handle HTTP error>
```

Reply:
Also, you don't need the `.new` - the exception will be created for you when you `raise`.

Reply:
```pry(main)> response = HTTParty.get("http://tinyurl.com/api-create.php?url=''")
=> "http://tinyurl.com/6mfcgml"```
You lost 'p' in 'php' before '?'. The two only errors I know of look like this.
```
[10] pry(main)> response = HTTParty.get("http://tinyurl.com/api-create.php?url=/")
=> "ERROR"
[11] pry(main)> response.code
=> 200
[12] pry(main)> response = HTTParty.get("http://tinyurl.com/api-create.php?url=")
=> "Error"
[13] pry(main)> response.code
=> 400
```

Reply:
OK, that will work.   What I was trying to get at is the situation where the website is down and/or the request times out - in that case we won't get an "Error" from the site, but we should get a response.code other than `200`.

So, maybe we should check for code `/error/I` _or_ code != `200`.